strawberryfields/apps/graph/resize.py

```diff
@@ -245,3 +236,28 @@ def greedy_degree(
 """dict[str, func]: Included methods for resizing subgraphs. The dictionary keys are strings
 describing the method, while the dictionary values are callable functions corresponding to the
 method."""
+
+
+def clique_shrink(subgraph: list, graph: nx.Graph) -> list:
+    """Shrinks an input subgraph until it forms a clique.
+
+    Args:
+        subgraph (list[int]): a subgraph specified by a list of nodes
+        graph (nx.Graph): the input graph
+
+    Returns:
+        list[int]: a clique of size smaller than or equal to the input subgraph
+    """
+
+    if not utils.is_subgraph(subgraph, graph):
+        raise ValueError("Input is not a valid subgraph")
+
+    subgraph = graph.subgraph(subgraph).copy()  # Create NetworkX Graph version of subgraph
+
+    while not utils.is_clique(subgraph):
+        degrees = list(subgraph.degree())
+        random.shuffle(degrees)
+        to_remove = min(degrees, key=lambda x: x[1])
```

Comment:
If it's a multiline comment, better to put it on a separate line above the function call.
```
# make sure selection of node with lowest degree are not deterministic in case of a tie
# https://docs.python.org/3/library/functions.html#min
np.random.shuffle(degrees)
```